packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -395,6 +395,36 @@ abstract class Widget extends DiagnosticableTree {
   }
 }
 
+// A cache of inherited elements with an optional parent pointer.
+//
+// When looking up an inherited element, this will look through parent
+// pointers until the element is found or the end is reached. When an element
+// is found, it is cached at the first element where the search began.
+//
+// The intention of this cache is to speed up the initial build of widget
+// trees that contain a significant number of inherited widgets by deferring
+// expensive map allocations until they are needed.
+class _InheritedLookup {
+  _InheritedLookup? parent;
```

Comment:
Can parent change? it's marked mutable here but there's no effort to invalidate the cache when it changes.

I don't see any code that changes it after the initial creation, which suggests maybe it should be final and set in the constructor?

Reply:
Done